### PR TITLE
Changes for create collection option names

### LINKED ISSUE #567: Vector options name change
Change the name for vector options in createCollection command:
1) `size` -> `dimension` and 
2) `function` -> `metric`

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
Changes for option names size -> dimension and function -> metric
**Which issue(s) this PR fixes**:
Fixes #567 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java [hunk 1]
```diff
@@ -41,12 +41,12 @@ public record Options(
...
     public record VectorSearchConfig(
-        @Positive(message = "size should be greater than `0`")
+        @Positive(message = "dimension should be greater than `0`")
             @Schema(
-                description = "Vector field embedding size",
+                description = "Dimension of the vector field",
                 type = SchemaType.INTEGER,
                 implementation = Integer.class)
-            Integer size,
+            Integer dimension,
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java:49]: I think combination of:

```
@JsonProperty("dimension")
@JsonAlias("size")
```

should work here, to support old name too (actually `@JsonProperty` is optional mostly for documentation, only alias needed)

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java [hunk 2]
```diff
@@ -57,10 +57,10 @@ public record VectorSearchConfig(
                 defaultValue = "cosine",
                 type = SchemaType.STRING,
                 implementation = String.class)
-            String function) {
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java:60]: And here similarly

```
@JsonProperty("metric")
@JsonAlias("function")
```

**Review by tatu-at-datastax — COMMENTED** [review]
Looks good, but I think we need `@JsonAlias` added to support old names.
And maybe just one test using old names as well?

**Review by tatu-at-datastax — APPROVED** [review]

**Review by vkarpov15 — APPROVED** [review]
LGTM :+1:

**Pull request merged**